Remove vendored asv

esc opened the pull request:
Replace the vendored `asv` with an install from `conda-forge`.

esc committed:
remove vendored asv (git submodule)

This submoudle stems from an older version of these benchmarks when ASV
wasn't yet ready. We can now install a package from the conda-forge
channel on anaconda.org so we no longer need the submodule.

esc committed:
update instructions for obtaining asv

We can now install asv from the conda-forge channel on anaconda.org

sklam commented on README.md:
````diff
@@ -13,13 +13,11 @@ submodule under `./asv`
 
 ```bash
-conda create -n numba-benchmark python=3.6
+conda create -c conda-forge -n numba-benchmark python=3.8 asv
````
should we be using the conda-forge build of python?

sklam replied:
alternative to use conda-forge only for asv
`conda create -n numba-benchmark python=3.8  conda-forge::asv`

seibert replied:
the `conda-forge::asv` form would be my vote, since it adds `conda-forge` to the solver as lower priority that `defaults`.  Putting just `-c conda-forge` will give it higher priority for all packages.

esc replied:
Installing `asv` will also pull in the following:

```
  asv                conda-forge/osx-64::asv-0.4.2-py38hc84c608_0
  python_abi         conda-forge/osx-64::python_abi-3.8-1_cp38
  six                conda-forge/noarch::six-1.15.0-pyh9f0ad1d_0
```

sklam replied:
`python_abi` is conda-forge's metapackage for abi.
`six` is fine.
i'm just worry that the environment starting with conda-forge's `python` will get mostly replaced when other dependency are installed, making the env less predictable

esc replied:
Thanks for the feedback on this, `conda-forge::asv` is what I went for.

esc committed:
update how we install the asv dependency

The `conda-forge::asv` form adds `conda-forge` to the solver as lower
priority that defaults. Putting just `-c conda-forge` will give it
higher priority for all packages.

esc commented:
@seibert @sklam all comments addressed, ready to merge?

stuartarchibald reviewed (approved):
Thanks for the updates, looks like all comments were addressed.

esc commented:
Thanks for merge.
